src/common/Requirements.ts

```diff
@@ -116,47 +118,52 @@ export abstract class BaseSetup implements RequirementList {
 
     constructor(logger: Logger) {
         this.logger = logger;
-        this.requirements = [];
-        this.additionalRequirements = [];
+        this.baseRequirements = { requirements: [], enabled: true };
+        this.commandRequirements = { requirements: [], enabled: true };
+        this.requirementsCheckFailureMessage = '';
+        this.requirementsCheckRecommendationMessage = '';
     }
 
     /**
      * Executes all of the base and additional requirement steps.
-     * @returns A SetupTestResult containing all of the results of the requirement tests
      */
-    public async executeSetup(): Promise<SetupTestResult> {
```

Comment:
Since we're not returning `SetupTestResult` anymore, then does the interface definition for it still needs to be using `export`? Same for `RequirementResult` interface

Reply:
I went with `RequirementCheckResult`.